### PR TITLE
[Code health] refactor: split localdatastore into subcomponents

### PR DESCRIPTION
**Pull request by scolsen**
Breaks down the monolithic datastore into a composition of a few dedicated data access components for each model object in the application. This  will enable us to update the implementations modularly or swap them out for alternatives. It should also help keep the diffs focused as we incrementally move to coroutines.

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/persistence/local/LocalDataStore.kt [hunk 1]
```diff
@@ -48,76 +43,18 @@ import io.reactivex.Single
...
-    submissionId: String
-  ): @Cold Maybe<Submission>
-
-  /**
-   * Returns a long-lived stream that emits the full set of tiles on subscribe and continues to
-   * return the full set each time a tile is added/changed/removed.
-   */
-  val tileSetsOnceAndStream: @Cold(terminates = false) Flowable<ImmutableSet<TileSet>>
+  /** Provides access to [Survey] data in local storage. */
+  var surveyStore: SurveyStore
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/persistence/local/LocalDataStore.kt:47]: If we're splitting up local data stores, for consistency would we then also split up remote ones? If so, should we prefix these with `Local` now to differentiate them conceptually from their eventual remote counterparts?

File: ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt [hunk 2]
```diff
@@ -172,8 +174,8 @@ constructor(
     )
 
   private fun applyAndEnqueue(mutation: SubmissionMutation): @Cold Completable =
-    localDataStore
-      .applyAndEnqueue(mutation)
+    submissionStore
+      .commitThenEnqueue(mutation)
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt:178]: Why the change from apply --> commit?

Also, because of the new files, git/GitHub aren't able to highlight which methods were renamed. Let's move the rename into its own PR or its own commit to make it clear?

**Review by gino-m — CHANGES_REQUESTED** [review]
LG, two questions/comments.

File: ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt (hunk 2 above)
Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/repository/SubmissionRepository.kt:178]: Ah, sorry about that -- yeah I can revert the rename and separate it out. The motivation was that kotlin has a stdlib scope function called `apply` (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/apply.html) so using `apply` in kotlin code requires ugly backticking.

File: ground/src/main/java/com/google/android/ground/persistence/local/LocalDataStore.kt (hunk 1 above)
Comment (scolsen) [reply ground/src/main/java/com/google/android/ground/persistence/local/LocalDataStore.kt:47]: sgtm! Will do

**Comment by scolsen** [comment]
@gino-m RFAL
